### PR TITLE
gd: enable AVIF support

### PR DESCRIPTION
**Pull request by malte-christian**
###### Motivation for this change

Enable AVIF support in GD. This fixes among other things AVIF image support in PHP 8.1.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/gd/default.nix [hunk 1]
```diff
@@ -9,6 +9,7 @@
 , libwebp ? null
 , libtiff ? null
 , libXpm ? null
+, libavif ? null
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/gd/default.nix:12]:
```suggestion
, libavif
```

Comment (malte-christian) [reply pkgs/development/libraries/gd/default.nix]: Removed all of them.

**Comment by SuperSandro2000** [comment]
Can you please target staging?

File: pkgs/development/libraries/gd/default.nix [hunk 2]
```diff
@@ -42,7 +43,7 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [ autoconf automake pkg-config ];
 
   buildInputs = [ zlib fontconfig freetype ];
-  propagatedBuildInputs = [ libpng libjpeg libwebp libtiff libXpm ];
+  propagatedBuildInputs = [ libpng libjpeg libwebp libtiff libXpm libavif ];
```
Comment (jtojnar) [inline pkgs/development/libraries/gd/default.nix:46]: Does this need to be propagated? `libavif` is in `Requires.private` in `${gd.dev}/lib/pkgconfig/gdlib.pc` so our patched `pkg-config` should should not need it to be propagated.

**Comment by malte-christian** [comment]
> Can you please target staging?

Changed base to staging and included a lot of unrelated commits by doing so. Rebased my commits now.

File: pkgs/development/libraries/gd/default.nix (hunk 2 above)
Comment (malte-christian) [reply pkgs/development/libraries/gd/default.nix:46]: A derivation with `libavif` placed within `buildInputs` seems to work as well. What about the other inputs which have already been propagated?

Comment (jtojnar) [reply pkgs/development/libraries/gd/default.nix:46]: It appears they have been introduced in https://github.com/NixOS/nixpkgs/commit/8b2fc35987e62c079fd50980782495360039a9c1. Not really much of context.

Comment (malte-christian) [reply pkgs/development/libraries/gd/default.nix]: Done

File: pkgs/development/libraries/gd/default.nix [hunk 3]
```diff
@@ -41,8 +42,8 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ autoconf automake pkg-config ];
 
-  buildInputs = [ zlib fontconfig freetype ];
-  propagatedBuildInputs = [ libpng libjpeg libwebp libtiff libXpm ];
+  buildInputs =
+    [ zlib fontconfig freetype libpng libjpeg libwebp libtiff libXpm libavif ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/gd/default.nix:46]:
```suggestion
  buildInputs = [ zlib fontconfig freetype libpng libjpeg libwebp libtiff libXpm libavif ];
```

**Review by SuperSandro2000 — APPROVED** [review]
Please squash merge when ofborg is done

**Comment by Artturin** [comment]
There's some nulls in all-packages.nix
```
gd = callPackage ../development/libraries/gd {
  automake = automake115x;
  libtiff = null;
  libXpm = null;
};
```

**Comment by SuperSandro2000** [comment]
#196037

**Pull request merged**